HashRegistrar.sol

```diff
@@ -148,17 +164,22 @@ contract Registrar {
     function startAuction(bytes32 _hash) noEther {
         entry newAuction = entries[_hash];
         if ((newAuction.status == Mode.Owned && now < newAuction.renewalDate) 
-            || (newAuction.status == Mode.Auction && now < newAuction.auctionExpiration))
+            || (newAuction.status == Mode.Auction && now < newAuction.registrationDate))
             throw;
         
         if (newAuction.status == Mode.Owned) {
             Deed deedContract = Deed(newAuction.deed);
             deedContract.closeDeed(999);
         }
         
-        newAuction.auctionExpiration = now + auctionLength;
+        // for the first six months of the registry, make longer auctions
+        // TODO: change 'hours' into 'weeks' for real contract
+        uint slowStart = 1 + (registryCreated + 20 hours - now) / 4 hours;
```

Comment:
You may be able to use a pattern where slowStart is read from a state variable. Said variable would be calculated in a separate function, pinged by kindly users who want their auctions to end sooner.

Having said that, the optimization may be so small that it's not worth the additional complexity.